REST API: Remove A/B test and update login flow

itsmeichigo opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #8845 
<!-- Id number of the GitHub issue this PR addresses. -->

## Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
This PR removes the A/B experiment for application password authentication and updates the login flow:
- Always show only the entry point to site address login on the prologue screen - WPCom login is hidden.
- When an address is entered, we do the following checks:
   - If the site is WPCom, show the WPCom login flow.
   - If the site is not WPCom but has Jetpack installed, activated and connected: show the WPCom login flow.
   - If the site is not WPCom and Jetpack is not installed, activated or connected: show the site credential login flow.

Unit tests have been updated accordingly. 

## Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->
**1. Log in to a WPCom site:**
- Log out of the app or skip onboarding if needed.
- On the prologue screen, notice that there are only 2 CTAs: Enter your site address and Get started.
- Tap Enter your site address and proceed with the address of your WPCom site.
- Notice that the WPCom email screen is displayed. Proceed to log in with your WPCom credentials.
- When the login completes, you should be navigated to the home screen.

**2. Log in to a self-hosted site with Jetpack:**
- Log out of the app or skip onboarding if needed.
- Tap Enter your site address and proceed with the address of your self-hosted site that has Jetpack installed and connected to a WPCom account.
- Notice that the WPCom email screen is displayed. Proceed to log in with your WPCom credentials.
- When the login completes, you should be navigated to the home screen.
- All features should be available: site switching, notifications, site visitors on the dashboard, password-protected product visibility.

**3. Log in to a self-hosted site without Jetpack.**
- Log out of the app or skip onboarding if needed.
- Tap Enter your site address and proceed with the address of your self-hosted site that doesn't have Jetpack installed, activated or connected to a WPCom account.
- Notice that the site credential screen is then displayed. Proceed to log in with your site credentials.
- When the login completes, you should be navigated to the home screen.
- Notice that a few features are missing:
   - Site visitors on the dashboard.
   - Site switching on the Menu tab.
   - Notifications for new orders or reviews.
   - Password-protected visibility for products.

## Screenshots
<!-- Include before and after images or gifs when appropriate. -->
N/A

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

itsmeichigo committed:
Remove track for applicationPasswordABTestVariation in AppCoordinator

itsmeichigo committed:
Remove check for applicationPasswordABTestVariation in AuthenticationManager and update login flow as planned

itsmeichigo committed:
Update tests for AuthenticationManager following the new login flow

itsmeichigo committed:
Remove AB test for application password authentication

itsmeichigo committed:
Remove analytics stats for REST API experiment

selanthiraiyan reviewed (commented):
>  Log in to a self-hosted site with Jetpack:

For the above scenario,

If I log in using site credentials, I am not asked for the WPCOM credentials. Is this expected? 


https://user-images.githubusercontent.com/524475/218383292-49fbc1ac-fa9d-48e6-b779-19482fd1d886.mp4

itsmeichigo committed:
Disable site credential login for jetpack sites

itsmeichigo committed:
Disable UITest cases for wpcom login if needed

itsmeichigo commented:
Thank you @selanthiraiyan for noticing the issue! As per the internal discussion on Slack [p1674806828177109-slack-C046HDZL87J] - I've added [a new PR](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/742) to hide the site credential login CTA on the WPCom email screen in the authenticator and updated this PR with the new change. Please test again to see if it works, thanks!

selanthiraiyan reviewed (approved):
Thank you for taking care of this @itsmeichigo 

This works great. 🚀

itsmeichigo committed:
Point WordPressAuthenticator to trunk

itsmeichigo committed:
Add accessibility ID for title label on login prologue page

itsmeichigo committed:
Expect only title label on prologue screen

itsmeichigo committed:
Update login flow with a new login method to switch to site address login if needed

itsmeichigo committed:
Update UI tests to use the new login method

itsmeichigo commented:
@woocommerce/mobile-ui-testing-squad I just had to update all the UI tests due to the change of CTAs on the prologue screen:
- Updated the expectation of the `PrologueScreen` to look for the title label on the login prologue pages rather than the CTA buttons. The reason is that these buttons are shown/hidden frequently based on our change of requirements, only the title labels stay.
- Added a new `login` method for the `LoginFlow`, where site address login is handled when WPCom login fails due to the missing of the WPCom login CTA.
- Updated all other tests to use the new `login` method.

Due to the urgency of the REST API project, it would be great if someone can review these changes soon before I can merge it for 12.4 code freeze. Thank you!

jostnes reviewed (approved):
> @woocommerce/mobile-ui-testing-squad I just had to update all the UI tests due to the change of CTAs on the prologue screen:
> 
> * Updated the expectation of the `PrologueScreen` to look for the title label on the login prologue pages rather than the CTA buttons. The reason is that these buttons are shown/hidden frequently based on our change of requirements, only the title labels stay.
> * Added a new `login` method for the `LoginFlow`, where site address login is handled when WPCom login fails due to the missing of the WPCom login CTA.
> * Updated all other tests to use the new `login` method.

Thank you for making the changes @itsmeichigo! Tested this out and the changes are good, all tests are passing in CI too. 

Considering that the prologue screen is changing quite a bit, the update to only look for the title makes sense. Approving on behalf of the mobile-ui-testing-squad 👍
